src/main/kotlin/com/strumenta/kolasu/model/Processing.kt

```diff
@@ -272,12 +272,133 @@ fun Node.mapTree(operation: (Node) -> Node): Node {
 }
 
 /**
- * Replace this node with "other" (by modifying the children of the parent node.)
- * For this to work, assignParents() must have been called.
+ * Replace [this] node with [other] (by modifying the children of the parent node.)
+ * For this to work, [Node.assignParents] must have been called.
  */
 fun Node.replaceWith(other: Node) {
     if (this.parent == null) {
         throw IllegalStateException("Parent not set")
     }
     this.parent!!.transformTree { if (it == this) other else it }
 }
+
+/**
+ * Looks for [oldNode] in the lists of nodes in this node.
+ * When found, it is removed, and in its place the [newNodes] are inserted.
+ */
+fun Node.replaceWithSeveral(oldNode: Node, newNodes: List<Node>) {
```

Comment:
Should we specify that in case a node is not contained in a list nothing will happen?

Reply:
If I'm not mistaken it will throw an exception. Shall I document that?

Reply:
Yes, please

Reply:
Done